server/src/main/java/org/opensearch/identity/noop/NoopPluginSubject.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.identity.noop;
+
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.identity.NamedPrincipal;
+import org.opensearch.identity.Subject;
+import org.opensearch.identity.tokens.AuthToken;
+import org.opensearch.threadpool.ThreadPool;
+
+import java.security.Principal;
+import java.util.concurrent.Callable;
+
+/**
+ * Implementation of subject that is always authenticated
+ * <p>
+ * This class and related classes in this package will not return nulls or fail permissions checks
+ *
+ * This class is used by the NoopIdentityPlugin to initialize IdentityAwarePlugins
+ *
+ * @opensearch.internal
+ */
+public class NoopPluginSubject implements Subject {
+    private final ThreadPool threadPool;
+
+    public NoopPluginSubject(ThreadPool threadPool) {
```

Comment:
```suggestion
    NoopPluginSubject(ThreadPool threadPool) {
```

Reply:
Changed, I introduced a separate ShiroPluginSubject for identity-shiro module which is the same implementation as this rendering it a Noop.